Astro 2.10 article (#757)
withastro/astro.build: src/content/blog/astro-2100.mdx

---
title: "Astro 2.10: Persistent State in View Transitions"
description: "Persistent UI in View Transitions and our first look at Astro 3.0"
publishDate: "August 3, 2023"
authors:
  - matthew
coverImage: "/src/content/blog/_images/astro-2100/blog-hero-2_10-release.png"
socialImage: "/src/content/blog/_images/astro-2100/og-image-2_10-release.png"
lang: "en"
---

Astro 2.10 is out with improvements to our experimental View transitions support. Additionally we've released our first early beta of Astro 3.0. Read on to learn more!

- **[Persistent UI in View transitions](#view-transitions-experimental)**

> - **[First Look at Astro 3.0](#astro-3)**

Author: Oops, I need to check both of these.

> - **[First Look at Astro 3.0](#astro-3)**

Author: Should be fixed.

If you already have Astro installed, you can upgrade it to 2.10 by running the upgrade command in your project (using your package manager of choice):

```
npm install astro@latest
pnpm upgrade astro --latest
yarn upgrade astro --latest
```

While you're at it, upgrade any `@astrojs/*` integrations and adapters you have installed, too!

# Persistent UI in View Transitions

> Our first release of the `<ViewTransition />` router allowed for smooth transitions between pages, but any stateful islands or elements you might have had would be rerendered.

Reviewer: ```suggestion
Our first release of the `<ViewTransition />` router brought smooth transitions between pages, but still caused any stateful islands or elements to be rerendered.
```

> In 2.10 we have fixed this with a new `transition:persist` directive.

Reviewer: ```suggestion
In 2.10, we've added a new `transition:persist` directive to the API to address this.
```
"fix" is a super strong claim at this stage of the game, maybe?  :sweat_smile:

> With this directive, you can keep the state of DOM elements and islands on the old page when transitioning to the new page. [Here's a demo](https://stackblitz.com/edit/github-stfdbe-ikkp2e?file=src%2Fpages%2Findex.astro) of it in action.

Reviewer: ```suggestion
With this directive, you can keep the state of DOM elements and islands from the old page when transitioning to the new page. [Here's a demo](https://stackblitz.com/edit/github-stfdbe-ikkp2e?file=src%2Fpages%2Findex.astro) of it in action.
```
This *could* be read as "keeping stuff ON the last page"... to rephrase like retaining the stuff FROM the last page when you go TO the new page is just less ambiguous!

For example, to keep a video playing across page navigation, add `transition:persist` to the element:

```astro
<video controls="" autoplay="" transition:persist>
  <source
    src="https://ia804502.us.archive.org/33/items/GoldenGa1939_3/GoldenGa1939_3_512kb.mp4"
    type="video/mp4"
  />
</video>
```

This `<video>` element, with its current state, will be moved over to the next page (if the video also exists on that page).

Likewise, this feature works with any client-side framework component island. In this example, a counter's state is preserved and moved to the new page:

```astro
<Counter count={5} client:load transition:persist />
```

See our [View Transitions Guide](https://docs.astro.build/en/guides/view-transitions/#maintaining-state) to learn more on usage.

## First Look at Astro 3.0

> We've released our first beta of Astro 3.0 as `astro@3.0.0-beta.0`. For those adventurous you can try out the release and see if you have compatibility issues with your app. So far this release contains mostly deprecations and removals of very old patterns. A few changes you might be interested in include:

Reviewer: ```suggestion
We've released our first beta of Astro 3.0 as `astro@3.0.0-beta.0`. If you're feeling adventurous, try it today! compatibility issues with your app. So far this release contains mostly deprecations and removals of very old patterns. However, you may experience some compatibility issues or breaking changes.

Some noteworthy changes include:
```

- HTML is now compressed by default.
- API routes now used capitalized method names, for example: `export function GET() ...`
- Adapters can inform Astro of which features they support, such as splitting entrypoints per page and Sharp / Squoosh compatibility.

[See the changelog](https://github.com/withastro/astro/blob/refs/heads/next/packages/astro/CHANGELOG.md#300-beta0) for a full list of the changes. We currently do not have documentation for the breaking changes but are working on a guide. As we approach 3.0 there will be more information about the changes.

> Once all of the breaking changes are complete there will be a release candidate and we hope to release the stable 3.0 in late August / early September.

Reviewer: ```suggestion
You can expect a release candidate once all of the breaking changes are complete, and we hope to release the stable 3.0 in late August / early September.
```

---
title: "Astro 2.10: Persistent State in View Transitions"
description: "Persistent UI in View Transitions and our first look at Astro 3.0"
publishDate: "August 3, 2023"
authors:
  - matthew
coverImage: "/src/content/blog/_images/astro-2100/blog-hero-2_10-release.png"
socialImage: "/src/content/blog/_images/astro-2100/og-image-2_10-release.png"
lang: "en"
---

Astro 2.10 is out with improvements to our experimental View transitions support. Additionally we've released our first early beta of Astro 3.0. Read on to learn more!

- **[Persistent UI in View transitions](#view-transitions-experimental)**
- **[First Look at Astro 3.0](#astro-3)**

If you already have Astro installed, you can upgrade it to 2.10 by running the upgrade command in your project (using your package manager of choice):

```
npm install astro@latest
pnpm upgrade astro --latest
yarn upgrade astro --latest
```

While you're at it, upgrade any `@astrojs/*` integrations and adapters you have installed, too!

## Persistent UI in View Transitions

> Our first release of the `<ViewTransition />` router brought smooth transitions between pages, but still caused any stateful islands or elements to be rerendered.

Reviewer 2: Comma in compound predicate isn't necessary, also missing hyphen in `rerendered`
```suggestion
Our first release of the `<ViewTransition />` router brought smooth transitions between pages but still caused any stateful islands or elements to be re-rendered.
```

In 2.10, we've added a new `transition:persist` directive to the API to address this.

With this directive, you can keep the state of DOM elements and islands from the old page when transitioning to the new page. [Here's a demo](https://stackblitz.com/edit/github-stfdbe-ikkp2e?file=src%2Fpages%2Findex.astro) of it in action.

For example, to keep a video playing across page navigation, add `transition:persist` to the element:

```astro
<video controls="" autoplay="" transition:persist>
  <source
    src="https://ia804502.us.archive.org/33/items/GoldenGa1939_3/GoldenGa1939_3_512kb.mp4"
    type="video/mp4"
  />
</video>
```

This `<video>` element, with its current state, will be moved over to the next page (if the video also exists on that page).

Likewise, this feature works with any client-side framework component island. In this example, a counter's state is preserved and moved to the new page:

```astro
<Counter count={5} client:load transition:persist />
```

See our [View Transitions Guide](https://docs.astro.build/en/guides/view-transitions/#maintaining-state) to learn more on usage.

## First Look at Astro 3.0

> We've released our first beta of Astro 3.0 as `astro@3.0.0-beta.0`. If you're feeling adventurous, try it today! compatibility issues with your app. So far this release contains mostly deprecations and removals of very old patterns. However, you may experience some compatibility issues or breaking changes.

Reviewer: ```suggestion
We've released our first beta of Astro 3.0 as `astro@3.0.0-beta.0`. If you're feeling adventurous, try it today! So far this release contains mostly deprecations and removals of very old patterns. However, you may experience some compatibility issues or breaking changes. Be warned, this early preview is changing rapidly!
```
